docs/products/add-data-to-grid-datastores/README.md

```diff
@@ -17,6 +17,37 @@ Datastores can be attached to Runs or Sessions and preserve the file format and
 We don't charge for data storage!
 :::
 
+## Datastores with Private S3 Buckets
+
+We only recommend using private S3 buckets for BYOC users. This can be accomplished with the following steps:
+
+1. follow the official aws [documentation](https://aws.amazon.com/premiumsupport/knowledge-center/cross-account-access-s3/).
+2. Use this for the principal field: "Principal": { "AWS": "arn:aws:iam::aws-account-id-associated-with-byoc:root" }
+3. Use the following policy to give access to all the contents within your bucket:
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::302180240179:root"
+            },
+            "Action": [
+                "s3:ListBucket",
+                "s3:GetObject",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": [
+               "arn:aws:s3:::your-bucket/*",
+               "arn:aws:s3:::your-bucket"
+           ]               
+        }
+    ]
+}
+
+### Known limitations:
+
+Currently, we cannot guarantee the privacy of datastores connected to a public grid cloud account. It is on the roadmap and will be implemented in a future release. However, if you would like to test this feature out you can replace the principal id with 302180240179.
```

Comment:
After second thought, I don't think we should say this publically

> However, if you would like to test this feature out you can replace the principal id with 302180240179

I might say something to the effect of: "there are limited circumstances under which testing of a private aws s3 bucket is possible on the grid public cloud. Please reach out to support@grid.ai if you are interested." that way we can make sure the user ACTUALLY understands the implications here before giving them an account ID which if they used would publically expose their data.

Reply:
addressed